gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/ConfigStoreBackedValueInspector.java

```diff
@@ -136,21 +127,40 @@ public Config getResolvedConfig(ConfigKeyPath configKey) {
     if(configKey.isRootPath()){
       return initialConfig;
     }
-    
+
     List<ConfigKeyPath> ownImports = this.topology.getOwnImports(configKey);
     // merge with other configs from imports
     if(ownImports!=null){
       for(ConfigKeyPath p: ownImports){
-        initialConfig = initialConfig.withFallback(this.getResolvedConfig(p));
+        initialConfig = initialConfig.withFallback(this.getResolvedConfigRecursive(p));
       }
     }
-    
+
     // merge with configs from parent for Non root
-    initialConfig = initialConfig.withFallback(this.getResolvedConfig(configKey.getParent())).resolve();
+    initialConfig = initialConfig.withFallback(this.getResolvedConfigRecursive(configKey.getParent()));
```

Comment:
This was a bug to resolve on every recursive call. The resolve has been moved to `getResolvedConfig` after all the recursive fallbacks are set. 
As per typesafe documentation, resolve should be called only once after all the fallbacks are set. 
See unit test `TestConfigStoreValueInspector.testResolveConfigOverridingInChild()` which fails if you resolve on every recursive call.